src/mapping/query/function/parse_duration.rs

```diff
@@ -0,0 +1,176 @@
+use super::prelude::*;
+use lazy_static::lazy_static;
+use regex::Regex;
+use rust_decimal::{prelude::ToPrimitive, Decimal};
+use std::collections::HashMap;
+use std::str::FromStr;
+
+lazy_static! {
+    static ref RE: Regex =
+        Regex::new(r"(?i)\A(?P<value>[0-9]*\.?[0-9]+)\s?(?P<unit>[a-z]{1,2})\z").unwrap();
```

Comment:
I think this regex is long enough to warrant a comment explaining what it does.

What does the `(?i)` at the start do?

Reply:
`(?i)` makes the whole regex case-insensitive (which in this case, admiditly means `[a-z]` at the end becomes `[a-zA-Z]`.

I'll add the `x` flag so that I can define the regex over multiple lines and add comments 👍

Reply:
This is also unable to parse more complex duration formats, such as `1:23s` or `1:22:33.456789`. Is that important?

Reply:
For now, the idea is to start with this syntax (including the optional space between the value and the unit). We can expand this as we get more requests for other formats.